testdrive/testdrive.go

```diff
@@ -146,15 +146,21 @@ Follow these instructions to create a token (we don't store any tokens):
 		colorstring.Println("[green]=> terraform found in $PATH![reset]")
 	}
 
-	// Download ngrok.
-	colorstring.Println("=> downloading ngrok  ")
-	s.Start()
-	ngrokURL := fmt.Sprintf("%s/ngrok-stable-%s-%s.zip", ngrokDownloadURL, runtime.GOOS, runtime.GOARCH)
-	if err = downloadAndUnzip(ngrokURL, "/tmp/ngrok.zip", "/tmp"); err != nil {
-		return errors.Wrapf(err, "downloading and unzipping ngrok")
+	// Detect ngrok and install it if not installed
+	_, ngrokErr := exec.LookPath("ngrok")
+	if ngrokErr != nil {
+		colorstring.Println("[yellow]=> ngrok not found in $PATH.[reset]")
+		colorstring.Println("=> downloading ngrok")
+		s.Start()
+		ngrokURL := fmt.Sprintf("%s/ngrok-stable-%s-%s.zip", ngrokDownloadURL, runtime.GOOS, runtime.GOARCH)
+		if err = downloadAndUnzip(ngrokURL, "/tmp/ngrok.zip", "/tmp"); err != nil {
+			return errors.Wrapf(err, "downloading and unzipping ngrok")
+		}
+		s.Stop()
+		colorstring.Println("[green]=> downloaded ngrok successfully![reset]")
+	} else {
+		colorstring.Println("[green]=> ngrok found in $PATH![reset]")
```

Comment:
nit: why not show the path of the executable since it has already been looked up?

Reply:
Done!